students/john_hunter/lesson03/mailroom.py

```diff
@@ -0,0 +1,191 @@
+            add_donor(input_name)
+            list_or_not = 'n'
+            more_donations = 'n'
+            break
+    mas_donations = 'Would you like to add more donations for ' + input_name + '? y/n: '
+    if list_or_not == 'y':
+        more_donations = input(mas_donations)
+    if more_donations == 'y':
+        totals = add_donations(input_name)
+    else:
+        totals = 0
+        for z in range(len(donors)):
+            if input_name == donors[z][0]:
+                for c in range(len(donors[z])-1):
+                    totals = totals + int(donors[z][c+1])
+                break
+
+
+    email_text_1 = "Dear " + input_name
+    email_text_2 = "Thank you for your generous donation(s) of $" + str(totals) + '.'
+    email_text_3 = "Sincerely," + '\n' + 'John Hunter'
+    print(email_text_1 + '\n'*2 + email_text_2 + '\n'*2 + email_text_3)
+
+def add_donor(name):
+    """Adds a donor to the donor dictionary and allows the user to add donation values
+
+    Returns: None
+    """
+    choice = 'y'
+    donations = list()
+    print('The name submitted is not on the list of donors.')
+    print('The donor name will be added, please add donation values: ')
+    while choice == 'y':
+        donation = int(input('Enter a donation amount, enter \'0\' to stop adding donation values: '))
+        if donation == 0:
+            choice = False
+        else:
+            donations.append(donation)
+    donations.insert(0, name)
+    new_donor = donations
+    donors.append(new_donor)
+    print('The new donor has been added:')
+    print(new_donor)
+    print('The Thank You email for the new donor is:')
+
+def order(total):
+    return sum(total[1:])
+
+def add_donations(name):
+    """
+    For either an existing donor or a new donor, we allow the user to add additional donations
+    input variable, 'name' allows the user to specify which key in the global dict has the
+    list values appended.
+    returns the sum of the donor's donations
+    """
+    donations = list()
+    donation = int()
+    while True:
+        donation = \
+        int(input('Enter a donation amount, enter \'0\' to stop adding donation values: '))
+        if donation == 0:
+            break
+        else:
+            int(donation)
+            donations.insert(-1, donation)
+    for i in range(len(donors)):
+        if name == donors[i][0]:
+            donors[i] = donors[i] + donations
+            break
+    print(donors[i])
+    return sum(donors[i][1:])
+
+def run_report():
+    """Returns a report of the donors by total historical amount
+    The report should contain the Donor Name, total donated, number of
+    donations, and average donation amount as values in each row
+   """
+    max_len_of_name = 0
+    donors_temp_two = donors
+
+    for i in range(len(donors_temp_two)):#calculates the longest name for formating
+        if len(donors_temp_two[i][0]) > max_len_of_name:
+            max_len_of_name = len(donors_temp_two[i][0])
```

Comment:
this is a bit overcomplicated :
```

for donor in donors:
    name = donor[0]
    if len(name) > max_len_of_name:
        max_len_of_name = name
```

no need for temp variable or using range() function.